lib/src/network/protocol/proxy_protocol/header.rs

```diff
@@ -130,36 +130,171 @@ pub struct HeaderV2 {
   ver_and_cmd: u8,     // protocol version and command
   family: u8,          // protocol family and address
   len: u16,            // number of following bytes part of the header
-  addr: AddrV2,
+  addr: ProxyAddr,
 }
 
 impl HeaderV2 {
-  fn new(ver_and_cmd: u8, family: u8, addr: AddrV2) -> Self {
+  pub fn new(addr_src: SocketAddr, addr_dst: SocketAddr) -> Self {
+    let addr = ProxyAddr::from(addr_src, addr_dst);
+
     HeaderV2 { 
       signature: [0x0D, 0x0A, 0x0D, 0x0A, 0x00, 0x0D, 0x0A, 0x51, 0x55, 0x49, 0x54, 0x0A],
-      ver_and_cmd,
-      family,
-      len: 0,
+      ver_and_cmd: 0x20,
+      family: get_family(&addr),
+      len: addr.len(),
```

Comment:
this field is not necessary since it's possible to do `self.addr.len()` (as done at line 161) instead of `self.len` (as done at line 154)

Reply:
This field is necessary when you parse a header coming from a client.

Reply:
is it really? It should be possible to read the length field, then use `take!()` with that length, and directly parse the rest of the data

Reply:
> is it really?

Yes, if you take a look at the [Haproxy src code](https://github.com/haproxy/haproxy/blob/25f067ccec52f53b0248a05caceb7841a3cb99df/src/connection.c#L987) they use a `DEFINE` for the address length part when they send a proxy protocol signature.
And for the [parsing](https://github.com/haproxy/haproxy/blob/25f067ccec52f53b0248a05caceb7841a3cb99df/src/connection.c#L527), they use the `len` field only to "secure" the parsing by checking if the len is correct. But they don't use the `len` field to [deserializes](https://github.com/haproxy/haproxy/blob/25f067ccec52f53b0248a05caceb7841a3cb99df/src/connection.c#L545) the signature received.
For the definition of our `header_v2` struct, I mostly copied their [struct proxy_hdr_v2](https://github.com/haproxy/haproxy/blob/25f067ccec52f53b0248a05caceb7841a3cb99df/include/types/connection.h#L332) where they keep the `len` field. Do you want to change that or it's okai and this `PR` is ok to be merge ?

Reply:
this length is used only for parsing, and its value depends on the other fields, so I'd rather have this field removed and the value generated from the other elements

Reply:
ack